### ISSUE TITLE
Message loss?

### ISSUE THREAD

**Issue by Laro88** [thomasnordquist/MQTT-Explorer#270]
I have had a couple of instances where MQTT messages are not showing up. It happens really seldom, but I accidentally managed to trigger it today - can't reproduce though.

I saw the expected message on MQTT.fx and simply did not see it in MQTT Explorer for some reason.

Are there any logfiles I can run through to see if there was a message being dropped or anything I can do do investigate.

I believe to have experienced a drop a couple of times i think, but just considered it a glitch in my code due to debugging or something - finally found a confirmed missing packet.

**Comment by thomasnordquist** [comment]
That a message was not received feels unlikely, that the "frontend" did not update afterwards appears more probable.

Did you receive another message on the same topic and the message was still not in the "History"?

I'll definitely check the receive/buffer-logic.

**Comment by Laro88** [comment]
Agreed, it was most likley reveived but the front end did not update with the topic visualized. 

I think I had 3-4 messages with the same topic and none of them visualized properly. Right now it is running smooth as usual.

The payload is a regular char* with a couple of different variants of human readable content and it normally works just fine, MQTT-fx is happy about it.

The message does not include a trailing null, and the reported message size seems ok.

**Comment by Laro88** [comment]
Just saw another curious thing Thomas.
I have two identical root topics in the MQTT explorer UI, never seen that one.
Right now I have around 100 messages pr second, seems to be running smooth.
![20191212 mqtt explorer duplicate topics](https://user-images.githubusercontent.com/12495700/70718124-eef4bf00-1cef-11ea-92df-59681c84237a.jpg)

Any chance of a race condition in the determination logic for "present or new key" in the tree component? I have not studied your code.

**Comment by thomasnordquist** [comment]
It certainly appears like it... This may in fact also be the reason a message might appear to be missing...
I'll see if I can reproduce it

**Comment by thomasnordquist** [comment]
1. Which mqtt-client are you using to publish this topic?
2. Which mqtt broker are you using?
3. Are the messages somehow special? (binary data for instance)

I'm currently flooding with ~700 text messages per second, totaling 500.000 until now. 
This could also be related to the web-framework I use (react), I'll try to provide you with a current build soon, to see if the problem persists at your end.

**Comment by Laro88** [comment]
1. paho-mqtt:x86-windows (vcpkg provided paho wrapper for c++)
2. mosquitto running in a docker container (seems to run flawlessly, clean logfile in docker)
3. Nothing binary (yet, we will shift some raw jpegs later)

I have a mqtt stresser running in docker as well, adding bogus mqtt workload did not take MQTT Explorer down.

**Comment by thomasnordquist** [comment]
I recently noticed message loss due to bad network. This should be rectifiable via QoS subscription options.
